src/dune_rules/dune_env.ml

```diff
@@ -201,7 +203,13 @@ module Stanza = struct
     and+ odoc = odoc_field
     and+ js_of_ocaml = js_of_ocaml_field
     and+ coq = coq_field
-    and+ format_config = Format_config.field ~since:(2, 8) in
+    and+ format_config =
+      Format_config.Generic.field ~since:(2, 8)
+      ~default_files:Ordered_set_lang.standard
+        ~files:
+          (field ~default:Ordered_set_lang.standard "files"
```

Comment:
This requires a version check since you're adding this feature to `(lang dune 3.7)`. I'd add a test for this too (in a project with lang 3.6, this should display an error asking to upgrade)